File: 0101-0200/0120.Triangle.cpp
//0ms


class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        int n=triangle.size();
        if(n==1) return triangle[0][0];

        for(int i=n-2;i>=0;i--)
        {
            for(int j=triangle[i].size()-1;j>=0;j--)
            {
                triangle[i][j]=triangle[i][j]+min(triangle[i+1][j+1],triangle[i+1][j]);
            }
        }
        return triangle[0][0];
    }
};

File: 0101-0200/0120.Triangle.rs
//0ms


impl Solution {
    pub fn minimum_total(mut triangle: Vec<Vec<i32>>) -> i32 {
        for y in (0..triangle.len() - 1).rev() {
            for x in (0..triangle[y].len()) {
                triangle[y][x] += triangle[y + 1][x].min(triangle[y + 1][x + 1]);
            }
        }
        triangle[0][0]
    }
}
